From dace8802ae6332eb4ca29faf288b96c83fadb2ae Mon Sep 17 00:00:00 2001
From: Arunachalam Ganapathy <arunachalam.ganapathy@arm.com>
Date: Tue, 28 Sep 2021 11:22:27 +0100
Subject: [PATCH 1/2] fix: fail mapping if start address exceeds page table
 mapping limit

Currently the end address is capped to stay within bounds of page
table range. If in case the start address is beyond bounds of
page table then no mapping is made but the mm_map_root incorrectly
returns success.

Signed-off-by: Arunachalam Ganapathy <arunachalam.ganapathy@arm.com>
Change-Id: I858ac653aafff62e9a2cf5771a32dc38690cd899
Upstream-Status: Pending [Not submitted to upstream yet]
---
 src/mm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/mm.c b/src/mm.c
index 6f15748..c57e24e 100644
--- a/src/mm.c
+++ b/src/mm.c
@@ -472,6 +472,10 @@ static bool mm_ptable_identity_map(struct mm_ptable *t, paddr_t pa_begin,
 	 */
 	CHECK(root_level >= 2);
 
+	if (begin >= ptable_end) {
+		return false;
+	}
+
 	/* Cap end to stay within the bounds of the page table. */
 	if (end > ptable_end) {
 		end = ptable_end;
-- 
2.30.2

